client/src/components/rooms/RoomsMeetingExperience.tsx

```diff
@@ -2,33 +2,228 @@
+  const [summarizationLanguage, setSummarizationLanguage] = useState<LocaleCode>('en-US');
+  const [summary, setSummary] = useState<SummarizeResult>();
+  const [summarizationStatus, setSummarizationStatus] = useState<'None' | 'InProgress' | 'Complete'>('None');
+  const [statefulClient, setStatefulClient] = useState<StatefulCallClient>();
+  const [callAgent, setCallAgent] = useState<DeclarativeCallAgent>();
+  const [call, setCall] = useState<Call | TeamsCall | undefined>();
+  const [callAdapter, setCallAdapter] = useState<CommonCallAdapter>();
+  const [customNotications, setCustomNotifications] = useState<ActiveNotification[]>([]);
+  const [callConnected, setCallConnected] = useState(false);
+  const [serverCallId, setServerCallId] = useState<string | undefined>(undefined);
+  const [transcriptionStartedByYou, setTranscriptionStartedByYou] = useState(false);
+
+  const transcriptionFeatureEnabled = transcriptionClientOptions?.transcription !== 'none';
+  const summarizationFeatureEnabled = transcriptionClientOptions?.summarization;
+  const theme = useTheme();
+  const callAutomationStarted = useRef(false);
+  const eventSourceRef = useRef<EventSource | null>(null);
+
+  useEffect(() => {
+    let eventSource: EventSource | null = null;
+
+    if (serverCallId && transcriptionFeatureEnabled) {
+      // Create EventSource connection when serverCallId is available. The URL provided here is for your server.
+      eventSource = new EventSource(`http://localhost:8080/api/notificationEvents`);
+      console.log(eventSource);
+      eventSourceRef.current = eventSource; // Store reference for cleanup
+
+      // Connection opened so we want to have the status of the transcription sent from the server to push the notification
+      eventSource.onopen = () => {
+        fetchTranscriptionStatus(serverCallId);
+      };
+      eventSource.addEventListener('message', (event) => {
+        console.log('EventSource connection event:', event);
+      });
+
+      // Connection error
+      eventSource.onerror = (error) => {
+        console.error('EventSource error:', error);
+
+        // Attempt to reconnect if the connection is closed
+        if (eventSource && eventSource.readyState === EventSource.CLOSED) {
+          console.log('EventSource connection closed, attempting to reconnect');
+        }
+      };
+    }
+
+    // Clean up on unmount or when serverCallId changes
+    return () => {
+      if (eventSource) {
+        console.log('Closing EventSource connection');
+        eventSource.close();
+      }
+    };
+  }, [serverCallId]);
+
+  useEffect(() => {
+    if (!eventSourceRef.current && !callConnected && transcriptionFeatureEnabled) {
+      return;
+    }
+    eventSourceRef.current?.addEventListener('TranscriptionStarted', (event) => {
+      const parsedData = JSON.parse(event.data);
+      if (parsedData.serverCallId.includes(serverCallId)) {
+        console.log('Transcription started', event.data);
```

Comment:
I noticed we have a bunch of console logs, these definitely can be useful but are all of them necessary?
For functions and events that do return something, or have some kind of visible action, does it need console logs?